packages/components/src/date-time/date.js

```diff
@@ -2,29 +2,86 @@
  * External dependencies
  */
 import moment from 'moment';
+import classnames from 'classnames';
+
 // react-dates doesn't tree-shake correctly, so we import from the individual
 // component here, to avoid including too much of the library
 import DayPickerSingleDateController from 'react-dates/lib/components/DayPickerSingleDateController';
 
 /**
  * WordPress dependencies
  */
-import { Component, createRef } from '@wordpress/element';
-import { isRTL } from '@wordpress/i18n';
+import { Component, createRef, useEffect, useRef } from '@wordpress/element';
+import { isRTL, _n, sprintf } from '@wordpress/i18n';
 
 /**
  * Module Constants
  */
 const TIMEZONELESS_FORMAT = 'YYYY-MM-DDTHH:mm:ss';
+const ARIAL_LABEL_TIME_FORMAT = 'dddd, LL';
+
+function DatePickerDay( { day, events = [] } ) {
+	const ref = useRef();
+
+	/*
+	 * a11y hack to make the `There is/are n events` string
+	 * available speaking for readers,
+	 * re-defining the aria-label attribute.
+	 * This attribute is handled by the react-dates component.
+	 */
+	useEffect( () => {
+		// Bail when no parent node.
+		if ( ! ref?.current?.parentNode ) {
+			return;
+		}
+
+		const { parentNode } = ref.current;
+		const dayAriaLabel = moment( day ).format( ARIAL_LABEL_TIME_FORMAT );
+
+		if ( ! events.length ) {
+			// Set aria-label without event description.
+			parentNode.setAttribute( 'aria-label', dayAriaLabel );
+			return;
+		}
+
+		const eventsString = sprintf(
+			/* translators: %d: number of calendar events. */
+			_n( 'There is %d event', 'There are %d events', events.length ),
+			events.length
+		);
+
+		parentNode.setAttribute(
+			'aria-label',
+			`${ dayAriaLabel }. ${ eventsString }`
```

Comment:
Is it possible to make this a single translation string instead of a concat on two translated strings together?

Translators get more context this way and it avoids a poor translation, especially if things would be phrased differently together.

Reply:
<3. I think it's handled here https://github.com/WordPress/gutenberg/pull/29716/commits/f677f377060999f700ac5e5311740afa5de4e5fa